microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Heap Snapshot Schema
+description: Use the Detached Elements tool in the Microsoft Edge DevTools to find and fix DOM memory leaks.
+author: MSEdgeTeam
+ms.author: joselea
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.date: 9/13/2022
+---
+# Heap Snapshot Schema
+
+_In this document we will explain the schema of a `*.heapsnapshot` file which you can take and export using DevTools. This is helpful in case you want to create your own analysis tools or visualizations._
+
+The DevTools Memory tool allows you to explore all the objects allocated in memory by taking a heap snapshot. This information is useful for performance investigations because you can find out what objects are consuming the most memory. Read more on [Heap Snapshots](heap-snapshots.md). 
+
+In DevTools the results are shown in the form of a table, in the top there is a list of aggerates that summarizes all the objects by type or class name and in the bottom there is a list of retainers which gets updated every time you elect a single object from the list of aggregates. 
+
+Memory in v8 is a graph made of nodes and edges. The purpose of the `*.heapsnapshot` file is to represent that graph efficiently and to facilitate its transfer in chunks between the browser process and DevTools, so the contents of this file are a flattened representation of those node-edge relations as simple arrays of numbers and strings. The file has a `*.heapsnapshot` extension but its contents are actually just a `JSON`. 
```

Comment:
This is probably for fairly advanced users anyway, but I wonder if we should give a short explanation about graphs, nodes and edges? Maybe with a little drawing too. Maybe a link to a wikipedia page like https://en.wikipedia.org/wiki/Graph_(abstract_data_type) .

Reply:
a link to the v8 doc could be better